### This is the assessment for Fundamentals of Data Analysis

Deadline/last commit on or before 29th Nov at https://github.com/kiyokosaito/Fundamentals-projects

What I will do on this project as follows.

### Description
    I will cover what overall 'Tips' data set is about in this description section. 
    
    Git hub URL : https://github.com/kiyokosaito/Fundamentals-projects
    
#### About Tips data set
'One waiter recorded information about each tip he received over a period of a few months working in one restaurant. He collected several variables' (https://rdrr.io/cran/regclass/man/TIPS.html)


The dataset contains a set of 244 records under 7 attributes -total_bill, tip, sex,	smoker,	day,	time and size(how many people in the party) 

#### Acknowledgements
This dataset is adapted from https://github.com/mwaskom/seaborn-data/blob/master/tips.csv





Data Sources - URL https://docs.google.com/spreadsheets/d/1Ihgub6a2AoSL9V0iCM6Zb6xoEaZnzK_pBlGTv6bTM2s/edit?usp=sharing

About this file
The dataset contains a set of 244 records under 7 attributes -total_bill, tip, sex, smoker, day, time and size(how many people in the party)

##### Columns = total_bill, tip,  sex, smoker, day,  time,	size (number of people in the party)

In [1]:
import pandas as pd

pd.read_csv("/Users/kiyok/Documents/tips.csv")

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

In [2]:
import numpy as np 

import matplotlib.pyplot as plt
%matplotlib inline

df = pd.read_csv("/Users/kiyok/Documents/tips.csv")

#this is to print average of total bill amount.
print((np.mean(df.iloc[:243,0])),'average of total bill') 



19.790082304526756 average of total bill


In [3]:
#this is to print average of tips 
print((np.mean(df.iloc[:243,1])),'average of tips') 

2.998271604938272 average of tips


In [4]:
#this is to print average of size of party 
print((np.mean(df.iloc[:243,6])),'average of size') 

2.57201646090535 average of size


In [5]:
# this is to print min and max of total bill and tips
print(min(df.iloc[:243,0]),'minimum of total bill')
print(max(df.iloc[:243,0]),'maximum of total bill')
print(min(df.iloc[:243,1]),'minimum of tips')
print(max(df.iloc[:243,1]),'maximum of tips')

3.07 minimum of total bill
50.81 maximum of total bill
1.0 minimum of tips
10.0 maximum of tips


 average of total bill = 19.79, min = 3.07, max = 50.81
 average of tips = 2.99, min = 1.0, max = 10
 average of size = 2.57
 

In [7]:

import pandas as pd
import numpy as np 

df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Thur.csv") 
df1 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Fri.csv") 
df2 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Sat.csv")
df3 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Sun.csv")
dff = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/female.csv") 
dfm = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/male.csv")
df13 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/1-3.csv")
df46 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/4-6.csv")
dfl = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/lunch.csv") 
dfd = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/dinner.csv")

#this is to print average of totall bills on Thur, Fri, Sat and Sun 
print((np.mean(df.iloc[:,0])),'average of total bills of Thur') 
print((np.mean(df1.iloc[:,0])),'average of total bills of Fri') 
print((np.mean(df2.iloc[:,0])),'average of total bills of Sat') 
print((np.mean(df3.iloc[:,0])),'average of total bills of Sun') 

#this is to print average of tips on Thur, Fri, Sat and Sun  
print((np.mean(df.iloc[:,1])),'average of tips of Thur') 
print((np.mean(df1.iloc[:,1])),'average of tips of Fri') 
print((np.mean(df2.iloc[:,1])),'average of tips of Sat') 
print((np.mean(df3.iloc[:,1])),'average of tips of Sun') 

#this is to print average of totall bills of female
print((np.mean(dff.iloc[:,0])),'average of total bills of female')

#this is to print average of totall bills of male 
print((np.mean(dfm.iloc[:,0])),'average of total bills of male')

#this is to print average of tips of female
print((np.mean(dff.iloc[:,1])),'average of tips of female')

#this is to print average of tips of male
print((np.mean(dfm.iloc[:,1])),'average of tips of male')

#this is to print average of totall bills of party size 1-3
print((np.mean(df13.iloc[:,0])),'average total bills of party size 1-3')
#this is to print average of totall bills of party size 4-6
print((np.mean(df46.iloc[:,0])),'average of total bills of party size 4-6')

#this is to print average of tips of party size 1-3
print((np.mean(df13.iloc[:,1])),'average of tips of party size 1-3')
#this is to print average of tips of party size 4-6
print((np.mean(df46.iloc[:,1])),'average of tips of party size 4-6')

#this is to print average of total bills of lunch
print((np.mean(dfl.iloc[:,0])),'average of totall bills of lunch')
#this is to print average of total bills of dinner
print((np.mean(dfd.iloc[:,0])),'average of totall bills of dinner')

#this is to print average of tips of lunch
print((np.mean(dfl.iloc[:,1])),'average of tips of lunch')
#this is to print average of tips of dinner
print((np.mean(dfd.iloc[:,1])),'average of tips of dinner')

17.807540983606557 average of total bills of Thur
17.785 average of total bills of Fri
20.44313953488371 average of total bills of Sat
21.468933333333332 average of total bills of Sun
2.7868852459016393 average of tips of Thur
2.8311111111111114 average of tips of Fri
2.995930232558138 average of tips of Sat
3.285066666666667 average of tips of Sun
18.069302325581393 average of total bills of female
20.810769230769246 average of total bills of male
2.8546511627906974 average of tips of female
3.098782051282052 average of tips of male
17.625177664974622 average total bills of party size 1-3
29.503777777777778 average of total bills of party size 4-6
2.723502538071067 average of tips of party size 1-3
4.257777777777777 average of tips of party size 4-6
17.243432835820894 average of totall bills of lunch
20.750457142857144 average of totall bills of dinner
2.7359701492537316 average of tips of lunch
3.103257142857143 average of tips of dinner


In [ ]:
# Specify total bills and tips 
X = df.iloc[:,[0,1]].values

# To make the scatter charts, nominate colors to each type of iris in 2 attributes.
# adapted from https://ameblo.jp/cognitive-solution/entry-12265630846.html 
plt.scatter(X[:243,0],X[:243,1],color='blue',marker='o')
plt.xlabel('total bills')
plt.ylabel('tips')
plt.show()

In [ ]:
#this is to print the total bills 
x =  df.iloc[:,[0]].values
plt.xlabel('total bill') 
plt.ylabel('number of people')
plt.hist(x)

In [ ]:
#this is to print tips 
x =  df.iloc[:,[1]].values
plt.xlabel('tips') 
plt.ylabel('number of people')
plt.hist(x)

## Regression 

These are the foundings :

1.majority of parties spent less than 20USD and tipped less than 4USD

2.the highest total bill was 50.81USD and the highest tip was 10USD

3.minimum tip was 1USD, this means every one who had something to eat at the restaurant paid tips

4.number of female was 87 and male was 157

5.151 were non smokers, 93 were smokers 

6.176 parties had dinner and 68 had lunch

7.this tips samples were taken from Thursday to Sunday with lunch or dinner time



In [ ]:
df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/male.csv") 
X = df.iloc[:,[0,1]].values
plt.scatter(X[:,0],X[:,1],color='blue',marker='o')
plt.xlabel('total bills') 
plt.ylabel('tips') 
plt.show() 

In [ ]:
df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/female.csv") 
X = df.iloc[:,[0,1]].values
plt.scatter(X[:,0],X[:,1],color='red',marker='o')
plt.xlabel('total bills') 
plt.ylabel('tips') 
plt.show() 

In [ ]:
# This is to visualise Total amount and tips based on the sex
# In order to create 2D chart, I chose female and male
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/female-male.csv") 

# Specify Total bills and tips values 
X = df.iloc[:,[0,1]].values

plt.scatter(X[:87,0],X[:87,1],color='red',marker='o',label='female')
plt.scatter(X[87:245,0],X[87:245,1],color='blue',marker='o',label='male')
plt.xlabel('Total bills')
plt.ylabel('Tips')
plt.legend(loc='upper left')
plt.show()

In [ ]:
df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/female-male.csv") 

# Specify Total bills and tips values based on days (Thur,Fri,Sat and Sun)
X = df.iloc[:,[0,1]].values

plt.scatter(X[:19,0],X[:19,1],color='purple',marker='o',label='Fri')
plt.scatter(X[20:106,0],X[20:106,1],color='green',marker='x',label='Sat')
plt.scatter(X[107:182,0],X[107:182,1],color='black',marker='^',label='Sun')
plt.scatter(X[183:244,0],X[183:244,1],color='yellow',marker='d',label='Thur')

plt.xlabel('Total bills')
plt.ylabel('Tips')
plt.legend(loc='upper left')
plt.show()


Collected 68 samples of lunches and saved as csv file and extract some interesting attributes from these samples 
Collected 176 samples of dinner and saved as csv file and extract some interesting attributes from these samples


In [ ]:
df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/lunch-dinner.csv") 

# Specify Total bills and tips values on Lunch and Dinner
X = df.iloc[:,[0,1]].values
plt.scatter(X[:176,0],X[:176,1],color='red',marker='s',label='Dinner')
plt.scatter(X[177:244,0],X[177:244,1],color='black',marker='s',label='Lunch')
plt.xlabel('Total bills')
plt.ylabel('Tips')
plt.legend(loc='upper left')
plt.show()

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/female.csv") 
df1 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/male.csv") 

X = df.iloc[:,0].values
Y = df.iloc[:,1].values

X1 = df1.iloc[:,0].values
Y1 = df1.iloc[:,1].values

#１つ目のプロット
plt.subplot(2,1,1)
plt.scatter(X,Y,color='red',marker='o',label='female')
plt.legend()
     

# 2つ目のプロット
plt.subplot(2,1,2)
plt.scatter(X1,Y1,color='blue',marker='o',label='male')
plt.legend()



In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/lunch.csv") 
df1 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/dinner.csv") 

X = df.iloc[:,0].values
Y = df.iloc[:,1].values

X1 = df1.iloc[:,0].values
Y1 = df1.iloc[:,1].values

#１つ目のプロット
plt.subplot(2,1,1)
plt.scatter(X,Y,color='black',marker='s',label='lunch')
plt.legend()
     

# 2つ目のプロット
plt.subplot(2,1,2)
plt.scatter(X1,Y1,color='red',marker='s',label='dinner')
plt.legend()



In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Thur.csv") 
df1 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Fri.csv") 
df2 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Sat.csv")
df3 = pd.read_csv("/Users/kiyok/Desktop/tips-data-set/Sun.csv")

X = df.iloc[:,0].values
Y = df.iloc[:,1].values

X1 = df1.iloc[:,0].values
Y1 = df1.iloc[:,1].values

X2 = df2.iloc[:,0].values
Y2 = df2.iloc[:,1].values

X3 = df3.iloc[:,0].values
Y3 = df3.iloc[:,1].values


#1st plot
plt.subplot(2,2,1)
plt.scatter(X,Y,color='yellow',marker='d',label='Thur')
plt.legend()
     

# 2nd plot
plt.subplot(2,2,2)
plt.scatter(X1,Y1,color='purple',marker='o',label='Fri')
plt.legend()

# 3rd plot
plt.subplot(2,2,3)
plt.scatter(X2,Y2,color='green',marker='x',label='Sat')
plt.legend()
     

# 4th plot
plt.subplot(2,2,4)
plt.scatter(X3,Y3,color='black',marker='^',label='Sun')
plt.legend()